File: cpp/leetcode/006.cpp
class Solution {
  public:
    string convert(string s, int numRows) {
        if (numRows == 1) {
            return s;
        }
        vector<string> res(numRows, "");
        for (int i = 0; i < s.size(); i++) {
            int idx = i % (2 * numRows - 2);
            int j = idx < numRows ? idx : 2 * numRows - 2 - idx;
            strs[j] += s[i];
        }
        string result = "";
        for (auto str : res) {
            result += str;
        }
        return result;
    }
};

File: python/leetcode/006.py
class Solution(object):
    def convert(self, s, numRows):
        if numRows == 1:
            return s
        strs = [""] * numRows
        for i, ch in enumerate(s):
            idx = i % (2 * numRows - 2)
            j = idx if idx < numRows else 2 * numRows - 2 - idx
            strs[j] += ch
        return "".join(strs)